### PR TITLE
Sort stations  and Online Users

### PR DESCRIPTION
**Pull request by thuytiendang206**

### CODE REVIEW

File: src/main/java/com/mgmtp/radio/controller/v1/StationController.java [hunk 1]
```diff
@@ -62,8 +67,8 @@ protected void initBinder(WebDataBinder binder) {
     })
     @GetMapping
     @ResponseStatus(HttpStatus.OK)
-    public Flux<StationDTO> getAllStation() {
-        return this.stationService.getAll();
+    public List<StationDTO> getAllStation() {
+            return this.stationService.getAllStationWithArrangement();
```
Comment (dungnv0811) [inline src/main/java/com/mgmtp/radio/controller/v1/StationController.java]: I think we should use getOrderedStations().

File: src/main/java/com/mgmtp/radio/controller/v1/StationController.java [hunk 2]
```diff
@@ -62,8 +64,8 @@ protected void initBinder(WebDataBinder binder) {
     })
     @GetMapping
     @ResponseStatus(HttpStatus.OK)
-    public Flux<StationDTO> getAllStation() {
-        return this.stationService.getAll();
+    public Map<String,StationDTO> getAllStation() {
```
Comment (dungnv0811) [inline src/main/java/com/mgmtp/radio/controller/v1/StationController.java]: should have one space after the comma

File: src/main/java/com/mgmtp/radio/controller/v1/StationController.java [hunk 3]
```diff
@@ -77,6 +79,11 @@ protected void initBinder(WebDataBinder binder) {
     @GetMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
     public Mono<StationDTO> getStation(@PathVariable(value = "id") String stationId) throws RadioNotFoundException {
+        if(getCurrentUser().isPresent()) {
```
Comment (dungnv0811) [inline src/main/java/com/mgmtp/radio/controller/v1/StationController.java]: should have one space after open "("

File: src/main/java/com/mgmtp/radio/sdo/Days_subtracted.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.mgmtp.radio.sdo;
+
+public enum Days_subtracted {
+    DAYS_SUBTRACTED(1L);
+    private  Long Days_limited;
```
Comment (dungnv0811) [inline src/main/java/com/mgmtp/radio/sdo/Days_subtracted.java]: one space before "Long"

File: src/main/java/com/mgmtp/radio/service/station/StationOnlineServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
+package com.mgmtp.radio.service.station;
+
+import com.mgmtp.radio.dto.station.StationDTO;
+import com.mgmtp.radio.dto.user.UserDTO;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service("stationOnlineService")
```
Comment (dungnv0811) [inline src/main/java/com/mgmtp/radio/service/station/StationOnlineServiceImpl.java]: just need @Service for short

File: src/main/java/com/mgmtp/radio/service/station/StationOnlineServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        if(!stationDTO1.isNewStation() && stationDTO2.isNewStation())
+            return SMALLER;
+        if(stationDTO1.getNumberOnline() > stationDTO2.getNumberOnline())
+            return LARGER;
+        if(stationDTO1.getNumberOnline() < stationDTO2.getNumberOnline())
+            return SMALLER;
+        return EQUAL;
+    };
+
+    public Map<String,StationDTO> getAllStation() {
```
Comment (dungnv0811) [inline src/main/java/com/mgmtp/radio/service/station/StationOnlineServiceImpl.java]: should be getAllStations

File: src/main/java/com/mgmtp/radio/service/station/StationOnlineServiceImpl.java (hunk 5 above)
Comment (thuytiendang206) [reply src/main/java/com/mgmtp/radio/service/station/StationOnlineServiceImpl.java]: Lam said, we write full like @Service("stationOnlineService") for clear

File: src/main/java/com/mgmtp/radio/sdo/DaySubtracted.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package com.mgmtp.radio.sdo;
+
+public enum DaySubtracted {
+    DAYS_SUBTRACTED(1L);
+    private Long DAYS_LIMITED;
```
Comment (dungnv0811) [inline src/main/java/com/mgmtp/radio/sdo/DaySubtracted.java]: should be: private final Long ONE_DAY_SUBTRACTED(1L)

File: src/main/java/com/mgmtp/radio/sdo/DaySubtracted.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
+package com.mgmtp.radio.sdo;
+
+public enum DaySubtracted {
+    DAYS_SUBTRACTED(1L);
```
Comment (dungnv0811) [inline src/main/java/com/mgmtp/radio/sdo/DaySubtracted.java]: should be: ONE_DAY_SUBTRACTED(1L);

**Pull request merged**